llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -132,6 +132,11 @@ cl::opt<bool> EnableMemProfContextDisambiguation(
 cl::opt<bool> SupportsHotColdNew(
     "supports-hot-cold-new", cl::init(false), cl::Hidden,
     cl::desc("Linking with hot/cold operator new interfaces"));
+
+cl::opt<bool> MemProfRequireDefinitionForPromotion(
+    "memprof-require-definition-for-promotion", cl::init(false), cl::Hidden,
```

Comment:
Should this flag default to true to preserve the current behaviour (when the flag is not set on the cmd line)?

The commit message also mentioned enabled by default.

Reply:
I meant to change it to be more aggressive by default and enable promotion with the target is a declaration (that's what I meant by "enabled" - promotion enabled, not the flag).

Reply:
Got it, thanks!

Reply:
I updated the description to be clearer about what I mean here, reread and it is definitely confusing